Record format validator

rowan04 opened the pull request:
- Creates validator django app
- Main validator page which accepts the input of an Apel record(s) and a record type (or all), and checks whether the record is valid or not.

Looks like:
<img width="645" height="568" alt="image" src="https://github.com/user-attachments/assets/ab57ed41-7195-4c32-b88c-83c7cd6cb77b" />

Resolves GT-1190
Working towards #47

rowan04 committed:
Create validator django app

- base file template for the new message validator
- makes use of django app creator startapp

rowan04 committed:
Update validator app config

- within apps.py
- update config to have the correct name

rowan04 committed:
Create initial html page template and validator view

- Linked into the Apel RecordFactory
- Doesn't make use of django model

rowan04 committed:
Ensure textarea input value is kept on form submission and refresh

- also cleans up some unnecessary code (unused exception handling)
- validation output is also kept on submission and refresh

rowan04 committed:
Add dropdown selection for record type

- has to import the apel record type classes
- has an option for all, which uses the record header instead

rowan04 committed:
Add improvement changes, including refactoring and improved styling

- adds more comments
- reworks exception handling
- adds apel packages to requirements
- adds styling
- adds link to validator page onto home monitoring page

rowan04 commented:
Oh I didn't realise the stylesheet.css file didn't already exist on GitHub. What is the reason for this?

tofu-rocketry commented:
> Oh I didn't realise the stylesheet.css file didn't already exist on GitHub. What is the reason for this?

It exists in the `publishing` app, so you shouldn't need to re-include it. https://github.com/apel/monitoring/blob/4f17c53e16f0c9a58c39c36d3009fb792e2f4fc0/monitoring/publishing/static/stylesheet.css

tofu-rocketry commented on monitoring/validator/views.py:
```diff
@@ -0,0 +1,90 @@
+    }
+
+
+    def validate(record: str, record_type: str) -> str:
```
Not keen on functions defined inside functions. Can this just go in the outer scope after the outer function?

tofu-rocketry commented on monitoring/templates/home.html:
```diff
@@ -42,5 +42,12 @@ <h2>Return format</h2>
 
+<h2>Record Validator</h2>
+<p><ul><li>
+    <a href="{% url 'validator' %}">{% url 'validator' %}</a> - A validating tool to check Apel records are
```
```suggestion
    <a href="{% url 'validator' %}">{% url 'validator' %}</a> - A validating tool to check APEL records are
```

tofu-rocketry commented:
See comments above. Also, can you rebase your branch please?

rowan04 committed:
Address code review comments on stylesheet and function location

- move function definition out of other function
- sort out stylesheet location error
- fix APEL typo

rowan04 commented:
> > Oh I didn't realise the stylesheet.css file didn't already exist on GitHub. What is the reason for this?
> 
> It exists in the `publishing` app, so you shouldn't need to re-include it. https://github.com/apel/monitoring/blob/4f17c53e16f0c9a58c39c36d3009fb792e2f4fc0/monitoring/publishing/static/stylesheet.css

I did as I believe you meant but my own environment was being very weird (i have 2 static folders somehow and it keeps picking up the other one) so I'm not sure if this works as expected or not, can you double check please?

tofu-rocketry reviewed (changes requested):
Looks better. Can you address the pre-commit warning?

rowan04 committed:
Fix failing pre-commit.ci check - blank lines at end of file

- had 2 blank lines instead of 1

rowan04 commented:
+ I have just remembered that this requires (or at least I did) an aquilon change (adding apel packages to the apel monitoring app template, `data-validation/config.pan`)

tofu-rocketry commented:
I'll hold this back to a `v1.7` so I can handle the Aquilon change separately.
